src/routes/solid-router/reference/data-apis/action.mdx

```diff
@@ -100,3 +101,69 @@ type Submission<T, U> = {
 const submissions = useSubmissions(action, (input) => filter(input));
 const submission = useSubmission(action, (input) => filter(input));
 ```
+##  Revalidate cached functions
+
+###  Revalidate all (default)
+By default all cached function will be revalidated wether the action does not return or return a "normal" response.
+```jsx
+
+const deleteTodo = action(async (formData: FormData) => {
+  const id = Number(formData.get("id"))
+  await api.deleteTodo(id)
+  // return new  Response("success",  {  status:  200  });
+  // ...
+})
+```
+
+### Revalidate specific cached keys
+
+By returning a response with solid router's `json` or `reload` helpers you can pass a key / keys to the revalidate prop as the second argument of solid's router json response helper.  You can either pass as string directly or use the cached functions `key` or `keyFor` props
+
+```jsx
+import { action, json } from "@solidjs/router"
+
+const deleteTodo = action(async (formData: FormData) => {
+  const id = Number(formData.get("id"))
+  await api.deleteTodo(id)
+  return json(
+  `deleted ${id}`,
+  { revalidate: ["getAllTodos",getTodos.key, getTodoByID.keyFor(id)]}
+  )
+  //or
+  return reload({ revalidate: ["getAllTodos",getTodos.key, getTodoByID.keyFor(id)]})
+})
+```
+
+### No revalidation
```

Comment:
Imho, "prevent" seems to convey better the message than simply "no" in this case.

```suggestion
Prevent revalidation
```